src/pages/patientView/mutation/VAFLineChartUtils.ts

```diff
@@ -20,16 +20,44 @@ function isPointBasedOnRealVAF(d: { mutationStatus: MutationStatus }) {
     );
 }
 
+function splitMutationsBySampleGroup(
+    mutations: Mutation[][],
+    sampleGroup: { [s: string]: string }
+) {
+    let groupedMutation: { [s: string]: Mutation[] } = {};
+    let groupedMutations: Mutation[][] = [];
+
+    mutations.forEach((mutation, i) => {
+        groupedMutation = {};
+        mutation.forEach((sample, j) => {
+            if (groupedMutation[sampleGroup[sample.sampleId]] === undefined) {
+                groupedMutation[sampleGroup[sample.sampleId]] = [];
+            }
+            //sample.groupByValue = sampleGroup[sample.sampleId];
+            groupedMutation[sampleGroup[sample.sampleId]].push(sample);
+        });
+        for (const m in groupedMutation) {
+            groupedMutations.push(groupedMutation[m]);
+        }
+    });
+    return groupedMutations;
+}
+
 export function computeRenderData(
     samples: Sample[],
     mutations: Mutation[][],
     sampleIdIndex: { [sampleId: string]: number },
     mutationProfileId: string,
-    coverageInformation: CoverageInformation
+    coverageInformation: CoverageInformation,
+    groupByOption: string,
```

Comment:
I get the list of clinical attributes for the Group By dropdown from sampleManager.getClinicalAttributeList as an array of clinical attribute id and clinical attribute displayName.
Once a clinical attribute is selected, the groupByOption will be set to the clinical attribute id (which is a string). But all logic is based on this id and never need to check its value.
The only value that I added and check is "None". So I can define a const GROUP_BY_NONE = "None" instead of enum, to not use the string "None" directly.

Reply:
oh I see. how about `groupByClinicalAttributeId` ? is that right?